packages/hash/shared/src/save.ts

```diff
@@ -345,8 +347,18 @@ const updateBlocks = defineOperation(
                     },
                   });
                 } else {
-                  // @todo check if the block data changed and push
-                  // an updateEntity action
+                  if (!isCreatedDraftEntity(draftBlockData)) {
+                    throw new Error(
+                      "Attempting to update block which has not yet been saved",
+                    );
+                  }
+                  updates.push({
+                    updateEntity: {
+                      entityId: draftBlockData.entityId,
+                      accountId: draftBlockData.accountId,
+                      properties: draftBlockData.properties,
+                    },
+                  });
```

Comment:
I assumed we'd want to check if the properties actually changed. Wouldn't this also push updateEntity action for blockData that didn't change?

Reply:
There's a check for that already above, unless I'm missing something?

Reply:
My bad. I forgot about that check.